### PR TITLE
Misc typing fixes

### PR DESCRIPTION
**Pull request by dstansby**
Typing fixes for a few more unrelated files.

### CODE REVIEW

File: napari/_event_loop.py [hunk 1]
```diff
@@ -8,5 +8,11 @@
     def gui_qt(**kwargs):
         raise exc
 
-    def run(**kwargs):
```
Comment (dstansby) [inline napari/_event_loop.py:11]: mypy expects the signatures of overriden functions to be exactly the same, hence the change.

File: pyproject.toml [hunk 2]
```diff
@@ -333,7 +331,6 @@ module = [
     'napari.layers.utils.string_encoding',
     'napari.layers.utils.style_encoding',
     'napari.layers.utils.text_manager',
-    'napari.resources._icons',
```
Comment (dstansby) [inline pyproject.toml:336]: There didn't seem to be any errors in this file.

File: napari/conftest.py [hunk 3]
```diff
@@ -283,7 +285,9 @@ def builtins(_npe2pm: TestPluginManager):
 @pytest.fixture
 def tmp_plugin(_npe2pm: TestPluginManager):
     with _npe2pm.tmp_plugin() as plugin:
-        plugin.manifest.package_metadata = {'version': '0.1.0', 'name': 'test'}
+        plugin.manifest.package_metadata = PackageMetadata(
+            version='0.1.0', name='test'
+        )
```
Comment (Carreau) [inline napari/conftest.py:290]: :+1:

**Review by Carreau — APPROVED** [review]

**Review by andy-sweet — APPROVED** [review]
Looks good. Thanks!

File: napari/_event_loop.py (hunk 1 above)
Comment (andy-sweet) [reply napari/_event_loop.py:11]: I was going to ask why the same isn't needed for `gui_qt`, but its original definition has the `@contextmanager` decorator so I guess this isn't strictly an override here even though the name is clobbered. Given that the `gui_qt` defined just raises, I think that's fine.

**Pull request merged**